### PR TITLE
Some simplficiations for the regs in mem ricsv machine

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

**Review by georgwiese — CHANGES_REQUESTED** [review]
Spelling error in the title

File: riscv/src/code_gen.rs [hunk 1]
```diff
@@ -964,10 +963,14 @@ fn process_instruction<A: InstructionArgs>(instr: &str, args: A) -> Result<Vec<S
                 vec![
                     format!("to_signed {}, {};", r1.addr(), tmp1.addr()),
                     // tmp2 is 1 if tmp1 is non-negative
-                    format!("is_positive {}, 0, 1, {};", tmp1.addr(), tmp2.addr()),
+                    format!(
+                        "is_diff_greater_than {}, 0, -1, {};",
```
Comment (georgwiese) [inline riscv/src/code_gen.rs:967]: Nice, that makes much more sense now, to test if it's non negative you test if it's greater than -1.

File: riscv/src/code_gen.rs [hunk 2]
```diff
@@ -1211,22 +1222,25 @@ fn process_instruction<A: InstructionArgs>(instr: &str, args: A) -> Result<Vec<S
...
+            vec![format!(
+                "branch_if_diff_equal {}, 0, 0, {label};",
+                r1.addr()
+            )]
         }
         "bgeu" => {
             let (r1, r2, label) = args.rrl()?;
             let label = escape_label(label.as_ref());
-            // TODO does this fulfill the input requirements for branch_if_positive?
+            // TODO does this fulfill the input requirements for branch_if_diff_greater_than?
```
Comment (georgwiese) [inline riscv/src/code_gen.rs:1241]: What does that even mean?

**Review by georgwiese — APPROVED** [review]
LGTM

File: riscv/src/code_gen.rs (hunk 2 above)
Comment (leonardoalt) [reply riscv/src/code_gen.rs:1241]: no idea, leftover from the very first implementation from @chriseth
